apps/files_sharing/lib/SharedStorage.php

```diff
@@ -105,14 +77,11 @@ public function __construct(array $parameters) {
 
 		parent::__construct([
 			'storage' => null,
-			'root' => null,
+			'root' => '',
 		]);
 	}
 
-	/**
-	 * @return ICacheEntry
-	 */
-	private function getSourceRootInfo() {
+	private function getSourceRootInfo(): ICacheEntry|false|null {
 		if (is_null($this->sourceRootInfo)) {
 			if (is_null($this->superShare->getNodeCacheEntry())) {
```

Comment:
```suggestion
		if (is_null($this->sourceRootInfo)) {
			$this->sourceRootInfo = $this->superShare->getNodeCacheEntry()
			if (is_null($this->sourceRootInfo)) {
```
This would avoid the else and calling the method twice